### ISSUE TITLE
Libwayland-client not found

### ISSUE THREAD

**Issue by ProbstDJakob** [wmww/gtk4-layer-shell#39]
With the release of the fedora package (#32) I've tried to use this library within my application. It worked, but with the problem _It works on my machine_ ^^. The problem is, that fedora only delivers the `libwayland-client.so` with the `wayland-devel` package and the non-development version `libwayland-client` does only deliver the `libwayland-client.so.0` and `libwayland-client.so.0.22.0` ([package files](https://packages.fedoraproject.org/pkgs/wayland/libwayland-client/fedora-40.html#files)) files. This has the effect that the code on [line 33](https://github.com/wmww/gtk4-layer-shell/blob/a689a399d37d174f2e93442e3adc931a9d8dee90/src/libwayland-shim.c#L33) successfully finds the library on my local machine where `wayland-devel` is installed, but not on the target system where only `libwayland-client` is installed.

Long story short, could the code on [line 33](https://github.com/wmww/gtk4-layer-shell/blob/a689a399d37d174f2e93442e3adc931a9d8dee90/src/libwayland-shim.c#L33) be extended to also look for `libwayland-client.so.0` (and maybe also for `libwayland-client.so.0.*` - other versions - if this is possible/needed)? If this is something which should be added I am willing to open a PR.

**Comment by wmww** [comment]
Hi, sorry for letting this sit for a while. That sounds very doable. Is this still an active issue?

**Comment by ProbstDJakob** [comment]
Yes, our current workaround is to install the development package to the production machines.

**Comment by wmww** [comment]
Can you give that a shot? If it's good I'll merge it and make a release.

**Comment by ProbstDJakob** [comment]
Is there a reason to not just letting the kernel load the shared library instead of using `dlopen`?

**Comment by wmww** [comment]
Does this answer your question? https://github.com/wmww/gtk4-layer-shell/blob/main/linking.md#underlying-cause

**Comment by wmww** [comment]
I think what I've got should be distro-agnostic? `*.0` indicates the 0th version of the ABI. It makes sense to link to and ship that in particular, since any other version is basically an entirely different library. I can't imagine why a distro wouldn't ship either `libwayland-client.so` or `libwayland-client.so.0`.

**Comment by ProbstDJakob** [comment]
This looks good. I would have over engineered it with support for looking for  `libwayland-client.so.0.*` as well, but this should work just fine I guess.
